projects/packages/forms/tests/php/contact-form/Feedback_Test.php

```diff
@@ -1181,6 +1181,66 @@ public function test_get_all_values_with_page_number() {
 		);
 	}
 
+	public function test_process_image_select_field_value() {
+		$current_post = Utility::create_post_context();
+		$form_id      = Utility::get_form_id();
+
+		// Create a form submission with two selected image choices
+		$_post_data = Utility::get_post_request(
+			array(
+				'images' => array(
+					'{"perceived":"B","selected":"B","label":"Test choice","showLabels":true,"image":{"id":null,"src":"https://www.example.com/test-choice.png"}}',
+					'{"perceived":"C","selected":"C","label":"Another test choice","showLabels":true,"image":{"id":12346,"src":"https://www.example.com/another-test-choice.png"}}',
+				),
+			),
+			'g' . $form_id
+		);
+
+		$form = new Contact_Form(
+			array(
+				'title'       => 'Test Form',
+				'description' => 'This is a test form.',
+			),
+			"[contact-field type='image-select' label='Images' isMultiple='1' options='A,B,C' showLabels='1' /]"
+		);
+
+		$response          = Feedback::from_submission( $_post_data, $form );
+		$feedback_post_id  = $response->save();
+		$saved_response    = Feedback::get( $feedback_post_id );
+		$saved_values      = $saved_response->get_all_values( 'submit' )['1_Images']['choices'];
+		$saved_values_json = array_map( 'json_encode', $saved_values );
+
+		Utility::destroy_post_context( $current_post );
+
+		$expected = array(
+			'type'    => 'image-select',
+			'choices' => array(
+				array(
+					'perceived'  => 'B',
+					'selected'   => 'B',
+					'label'      => 'Test choice',
+					'showLabels' => true,
+					'image'      => array(
+						'id'  => null,
+						'src' => 'https://www.example.com/test-choice.png',
+					),
+				),
+				array(
+					'perceived'  => 'C',
+					'selected'   => 'C',
+					'label'      => 'Another test choice',
+					'showLabels' => true,
+					'image'      => array(
+						'id'  => 12346,
+						'src' => 'https://www.example.com/another-test-choice.png',
+					),
+				),
+			),
+		);
+
+		$this->assertEquals( $expected, Feedback::process_image_select_field_value( $saved_values_json ), 'Processed image select field value should match the expected values' );
+	}
```

Comment:
The tests pass, but in this test we do a lot of work to get $saved_values_json, and then pass that back into the same function we're in. Can you give a bit more clarity on the logic or why we're doing that?

Reply:
You are right, I was trying to test a specific function by testing the whole submission process. Since the submission process includes the call to that specific function, I went ahead and removed the call to it and the manual `$saved_values_json` processing, now the test is more in line with what was done with the file upload field.